src/spatial_ref/srs.rs

```diff
@@ -451,6 +451,141 @@ impl SpatialRef {
             None
         }
     }
+
+    /// Get spheroid semi-major axis.
+    ///
+    /// Returns [`Err`] variant in case the C library returns an error code different from [`OGRERR_NONE`](https://gdal.org/api/vector_c_api.html#c.OGRERR_NONE).
+    ///
+    /// See: [`OSRGetSemiMajor`](https://gdal.org/api/ogr_srs_api.html#_CPPv415OSRGetSemiMajor20OGRSpatialReferenceHP6OGRErr)
+    pub fn semi_major(&self) -> Result<f64> {
+        let mut rv = OGRErr::OGRERR_NONE;
+        let a = unsafe { gdal_sys::OSRGetSemiMajor(self.0, &mut rv as *mut u32) };
+        if rv != OGRErr::OGRERR_NONE {
+            return Err(GdalError::OgrError {
+                err: rv,
+                method_name: "OSRGetSemiMajor",
+            });
+        }
+        Ok(a)
+    }
+
+    /// Get spheroid semi-minor axis.
+    ///
+    /// Returns [`Err`] variant in case the C library returns an error code different from [`OGRERR_NONE`](https://gdal.org/api/vector_c_api.html#c.OGRERR_NONE).
+    ///
+    /// See: [`OSRGetSemiMinor`](https://gdal.org/api/ogr_srs_api.html#_CPPv415OSRGetSemiMinor20OGRSpatialReferenceHP6OGRErr)
+    pub fn semi_minor(&self) -> Result<f64> {
+        let mut rv = OGRErr::OGRERR_NONE;
+        let b = unsafe { gdal_sys::OSRGetSemiMinor(self.0, &mut rv as *mut u32) };
+        if rv != OGRErr::OGRERR_NONE {
+            return Err(GdalError::OgrError {
+                err: rv,
+                method_name: "OSRGetSemiMinor",
+            });
+        }
+        Ok(b)
+    }
+
+    /// Set a projection parameter value.
+    ///
+    /// Returns [`Err`] variant in case the C library returns an error code different from [`OGRERR_NONE`](https://gdal.org/api/vector_c_api.html#c.OGRERR_NONE).
+    ///
+    /// See: [`OSRSetProjParm`](https://gdal.org/api/ogr_srs_api.html#_CPPv414OSRSetProjParm20OGRSpatialReferenceHPKcd)
+    pub fn set_proj_param(&mut self, name: &str, value: f64) -> Result<()> {
+        let c_name = CString::new(name)?;
+        let rv = unsafe { gdal_sys::OSRSetProjParm(self.0, c_name.as_ptr(), value) };
+        if rv != OGRErr::OGRERR_NONE {
+            return Err(GdalError::OgrError {
+                err: rv,
+                method_name: "OSRSetProjParm",
+            });
+        }
+        Ok(())
+    }
+
+    /// Fetch a projection parameter value.
+    ///
+    /// Returns [`Err`] variant in case the C library returns an error code different from [`OGRERR_NONE`](https://gdal.org/api/vector_c_api.html#c.OGRERR_NONE).
+    ///
+    /// See: [`OSRGetProjParm`](https://gdal.org/api/ogr_srs_api.html#_CPPv414OSRGetProjParm20OGRSpatialReferenceHPKcdP6OGRErr)
+    pub fn get_proj_param(&self, name: &str) -> Result<f64> {
```

Comment:
So ideally, this would return something like `Result<Option<f64>>` and the user could handle the missing case. But since we can do that, should we take the default as an argument?

It's currently ambiguous. Are the parameters never 0 in practice?

Reply:
Behavior of `OSRGetProjParm` is quite tricky, a least from Rust point of view.

In C and C++ if value is  found it is returned normally, but if value is not found, a value from default parameter is returned, but also error code (if pointer is not null) is set to `OGRERR_FAILURE`.  However looking from Rust point of view it may mean not found (therefore is candidate for `Option::None`) or can mean any other failure.
You can see this in GDAL [source code for `OSRGetProjParm`](https://github.com/OSGeo/gdal/blob/51d91973656df7be083a9194780da286c6a2cab7/ogr/ogrspatialreference.cpp#L5667).

To sum up:
* on existing value: we receive the value itself,
* if value does not exist: we receive default value and very general error code.

That's why I used only `Result<f64>` as return type and I also created also `get_proj_param_or_default`.

Reply:
Yeah, I'm inclined towards something like a single `pub fn get_proj_param(&self, name: &str, default: f64) -> Result<Option<f64>>`, which returns `Err` on a string conversion error (basically never), and `Ok(None)` on `OGRERR_FAILURE`. It's not that awkward to use, and should be pretty clear.

Both error cases in the implementation are similar, and boil down to "there's no parameter with that name on the `PROJCS`.